packages/notched-outline/index.tsx

```diff
@@ -91,49 +89,44 @@ export default class NotchedOutline extends React.Component<
     return classnames('mdc-notched-outline', Array.from(classList), className);
   }
 
-  get adapter() {
+  get adapter(): MDCNotchedOutlineAdapter {
     return {
-      getWidth: () => this.outlineElement_.current ? this.outlineElement_.current.offsetWidth : 0,
-      getHeight: () => this.outlineElement_.current ? this.outlineElement_.current.offsetHeight : 0,
       addClass: (className: string) =>
         this.setState({classList: this.state.classList.add(className)}),
       removeClass: (className: string) => {
         const {classList} = this.state;
         classList.delete(className);
         this.setState({classList});
       },
-      setOutlinePathAttr: (value: string) => {
-        if (this.pathElement_.current) {
-          this.pathElement_.current.setAttribute('d', value);
-        }
+      setNotchWidthProperty: (width: number) => {
+        this.setState({
+          actualWidth: width,
+        });
       },
-      getIdleOutlineStyleValue: (propertyName: string) => {
-        if (!this.idleElement_.current) return;
-        return window
-          .getComputedStyle(this.idleElement_.current)
-          .getPropertyValue(propertyName);
+      removeNotchWidthProperty: () => {
+        this.setState({
+          actualWidth: null,
+        });
       },
     };
   }
 
   render() {
     return (
-      <React.Fragment>
-        <div
-          className={this.classes}
-          key='notched-outline'
-          ref={this.outlineElement_}
-        >
-          <svg focusable='false'>
-            <path ref={this.pathElement_} className='mdc-notched-outline__path' />
-          </svg>
-        </div>
-        <div
-          ref={this.idleElement_}
-          className='mdc-notched-outline__idle'
-          key='notched-outline-idle'
-        />
-      </React.Fragment>
+      <div
```

Comment:
Not sure what otherProps users would add, but please add a way to pass properties here.